functions/src/on-create-bet.ts

```diff
@@ -26,8 +39,99 @@ export const onCreateBet = functions.firestore
       .update({ lastBetTime, lastUpdatedTime: Date.now() })
 
     await notifyFills(bet, contractId, eventId)
+    await updateUniqueBettorsAndGiveCreatorBonus(contractId, eventId)
   })
 
+const updateUniqueBettorsAndGiveCreatorBonus = async (
+  contractId: string,
+  eventId: string
+) => {
+  const userContractSnap = await firestore
+    .collection(`contracts`)
+    .doc(contractId)
+    .get()
+  const contract = userContractSnap.data() as Contract
+  if (!contract) {
+    log(`Could not find contract ${contractId}`)
+    return
+  }
+  const contractBets = (
+    await firestore
+      .collection(`contracts/${contractId}/bets`)
+      .where('userId', '!=', contract.creatorId)
+      .get()
+  ).docs.map((doc) => doc.data() as Bet)
```

Comment:
Seems like this won't scale well. Fetching all bets (except for the creator) on every bet is n^2.

Can't you just check if the new bet is a new unique bettor compared to the saved uniqueBettors field? If the uniqueBettors field doesn't exist then you could fetch all the bets.